manual/node_pattern.md

```diff
@@ -0,0 +1,270 @@
+# Node Pattern
+
+Node pattern is a small language to help find specific nodes in
+the Abstract Syntax Tree using a simple string.
+
+It reminds the simplicity of regular expressions but used to find specific
+nodes of ruby code.
+
+## Simple match
+
+A simple integer like `1` represents `(int 1)` in the AST.
+
+```
+$ ruby-parse -e '1'
+(int 1)
+```
+
+- `int` will match exactly the node
+- `(int 1)` will match precisely the node
+
+## `_` for any value
+
+`_` will check if there's something present in the specific position, no matter the
+value:
+
+- `(int _)` will match any number
+- `(int _ _)` will not match because `int` types have just one child that
+  contains the value.
+
+The compiler translates the pattern `(int _)` into something like:
+
+```ruby
+node.int_type? && node.children.size == 1
+```
+
+while `(int _ _)` will be translated to something like:
+
+```ruby
+node.int_type? && node.children.size == 2
+```
+
+## `...` for something else
+
+While `_` limits the size, `...` make it open for more children.
+
+And the example `(int ...)` will be translated into something like:
```

Comment:
`int` is definitely a bad example here, since there's just a single child node. `send` with couple parameters?